server/rpc/yorkie_server.go

```diff
@@ -460,21 +458,19 @@ func (s *yorkieServer) RemoveDocument(
 	}
 
 	project := projects.From(ctx)
-	if pack.HasChanges() {
-		locker, err := s.backend.Coordinator.NewLocker(ctx, packs.PushPullKey(project.ID, pack.DocumentKey))
-		if err != nil {
-			return nil, err
-		}
+	locker, err := s.backend.Coordinator.NewLocker(ctx, packs.PushPullKey(project.ID, pack.DocumentKey))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Change RemoveDocument locker condition (pack.HasChanges() => always)

If Client SDK executes `AttachDocument` and `RemoveDocument` at the same time, the following situation may occur. 
- case 1. `AttachDocument` => `RemoveDocument`: Run successfully
- case 2. `RemoveDocument` => `AttachDocument`: Failed to execute

The status of document changes as follow. (document-removal desing document)
An `Attached` state is only can be converted to `Removed`, but not allow to converted `Attached` to `Removed`
https://github.com/yorkie-team/yorkie/blob/main/design/document-removal.md#state-transition-of-document
```
 ┌──────────┐ Attach ┌──────────┐ Remove ┌─────────┐
 │ Detached ├───────►│ Attached ├───────►│ Removed │
 └──────────┘        └─┬─┬──────┘        └─────────┘
           ▲           │ │     ▲
           └───────────┘ └─────┘
              Detach     PushPull
```
This time, `AttachDocument` is modified to always have a lock, but if `RemoveDocument` is executed at the same time without a lock, it can cause situations that shouldn't happen above. To ensure that the two methods are always executed sequentially, `RemoveDocument` must also be executed with a locker.